app/models/User.js

```diff
@@ -251,13 +261,23 @@ schema.virtual('name').get(function () {
     return this.first + " " + this.last;
 });
 
-
+/**
+ * Check if email is taken
+ * @param {string} email - The user's email
+ * @param {ObjectId} [excludeUserId] - The id of the user to be excluded
+ * @returns {Promise<boolean>}
+ */
+ schema.statics.isEmailTaken = async function (email, excludeUserId) {
+    const user = await this.findOne({ email, _id: { $ne: excludeUserId } });
```

Comment:
We need to add a check to see if the email is active. A soft-deleted email would still exist in the database and the user wouldn't be able to register accordingly.
That email should be considered as "non-existent"